src/importer.cpp

```diff
@@ -714,6 +714,11 @@ void updateComponentsVariablesUnitsNames(const std::string &name, const Componen
 {
     for (size_t variableIndex = 0; variableIndex < component->variableCount(); ++variableIndex) {
         auto variable = component->variable(variableIndex);
+        if (component->isImport()) {
+            auto importModel = component->importSource()->model();
+            auto importComponent = importModel->component(component->importReference());
+            variable = importComponent->variable(variable->name());
+        }
```

Comment:
Might want to replace with something like:
```c++
VariablePtr variable;
if (component->isImport()) {
    auto importModel = component->importSource()->model();
    auto importComponent = importModel->component(component->importReference());
    variable = importComponent->variable(variable->name());
} else {
    variable = component->variable(variableIndex);
}
```

Reply:
I think I prefer the current implementation.

Reply:
To compute `component->variable(variableIndex);` every time is a waste of time when you have a lot of imports, hence my suggestion.

Reply:
Oops, I hadn't noticed the call to `variable->name()` in the case of an import. All good then!